dev/jenkins_task_validate_image.sh

```diff
@@ -0,0 +1,624 @@
+  run_standard_aws_test "aws_smoke_test"
+
+  # In case we are not PARALLELIZE_TESTS
+  wait_on_tests
+}
+
+
+function collect_reports_and_logs() {
+  # Render the journals into HTML
+  # (Individual HTML files and an overall index.html)
+  python -m citest.reporting.generate_html_report *.journal;
+  mkdir -p citest_logs
+  mv *.log citest_logs
+  mv *.out citest_logs
+  mv *.journal citest_logs
+
+
+  # Grab the log files from the instance before we succeed or fail
+  # The logs we grab here could help diagnose errors
+  echo "Collecting log files from server..."
+  local count=0
+  while ! compress_server_logs; do
+    ((count++)) || true
+    if [[ count -eq 60 ]]; then
+      echo "Giving up fetching logs."
+      break
+    fi
+    echo "Failed to fetch server logs ... try again."
+    sleep 1
+  done
+
+  gcloud compute copy-files $MY_INSTANCE_NAME:logs.tz . \
+     --account $TESTER_SERVICE_ACCOUNT \
+     --project $MY_SPINNAKER_PROJECT \
+     --zone $MY_ZONE
+
+  mkdir -p server_logs
+  pushd server_logs
+  tar xzf ../logs.tz
+  rm -f ../logs.tz
+  mv syslog syslog.log
+  popd
+}
+
+
+function archive_to_google_cloud_storage() {
+  local archive_dir=${BASE_ARCHIVE_PATH}/${BUILD_NUMBER}
+  gsutil cp citest_logs/*.journal ${archive_dir}
+  gsutil cp test_logs/*.snapshot-${BUILD_NUMBER} ${archive_dir}
+}
+
+
+function generate_cleanup_leaked_resource_script() {
+  local taskname=$(basename `pwd`)
+
+  cat > cleanup_leaks.sh <<EOF
+#!/bin/bash
+  if [[ \${#} -eq 0 ]]; then
+    BUILD_NUMBER=${BUILD_NUMBER}
+  else
+    BUILD_NUMBER=\$1
+  fi
+  echo "Cleaning up BUILD_NUMBER=\${BUILD_NUMBER}"
+
+  archive_dir=~jenkins/jobs/${taskname}/builds/\${BUILD_NUMBER}/archive
+  snapshot_dir=\${archive_dir}/test_logs
+
+  cd `pwd`
+  source testenv/bin/activate
+  for snapshot in container__${MY_SPINNAKER_PROJECT} \\
+                  compute_storage__${MY_SPINNAKER_PROJECT} \\
+                  compute_storage__spinnaker-build; do
+      echo "SNAPSHOT=\$snapshot"
+      if [[ -f \${snapshot_dir}/final__\${snapshot}.snapshot-\${BUILD_NUMBER} ]]; then
+          snapshot_name=final
+      elif [[ -f \${snapshot_dir}/checkpoint__\${snapshot}.snapshot-\${BUILD_NUMBER} ]]; then
+          snapshot_name=checkpoint
+      else
+          echo "No snapshot for '\${snapshot}' to compare against."
+          continue
+      fi
+      python citest/citest/gcp_testing/resource_snapshot.py \\
+          --compare \${snapshot_dir}/baseline__\${snapshot}.snapshot-\${BUILD_NUMBER} \\
+                    \${snapshot_dir}/\${snapshot_name}__\${snapshot}.snapshot-\${BUILD_NUMBER} \\
+          --exclude compute.*Operations,storage.*AccessControls \\
+                    container compute storage \\
+          --credentials_path=$PROJECT_JSON_PATH \\
+          --delete_added \\
+          --delete_for_real
+  done
+EOF
+  chmod +x cleanup_leaks.sh
+}
+
+######################################################################
+
+cleanup_workspace
```

Comment:
Do you think this should be wrapped in a function too?
If the whole thing wraps into functions with no side effects you can source it to get the definitions then run main() yourself or create your own flow out of the steps you want.